xlators/protocol/server/src/server.c

```diff
@@ -1711,6 +1727,23 @@ server_notify(xlator_t *this, int32_t event, void *data, ...)
             }
             GF_FREE(victim_name);
             break;
+        case GF_EVENT_SIGHUP:
+            if (conf->volfile_dir) {
+                pthread_mutex_lock(&conf->mutex);
+                {
+                    list_for_each_entry(xprt, &conf->xprt_list, list)
+                    {
+                        /* TODO: optimize by sending signal to only those who
+                         * fetched data once */
+                        rpcsvc_callback_submit(conf->rpc, xprt,
```

Comment:
It would probably be better to do this outside the mutex.

Reply:
checked and the code to send even the lock notifications are inside of the mutex... 

the challenge is iterating on the rpc, which may be altered, lock is required imo. Havent changed here for now, as it is working. Would love to hear suggestions on how we can move it out of the locked region.

Reply:
Ok